README.md

```diff
@@ -502,6 +503,38 @@ Supported error codes:
 * `pb_utils.TritonError.UNAVAILABLE`
 * `pb_utils.TritonError.UNSUPPORTED`
 * `pb_utils.TritonError.ALREADY_EXISTS`
+* `pb_utils.TritonError.CANCELLED` (since 23.10)
```

Comment:
We will also need to update for the response factory a method for determining if the associated request is cancelled (To handle decoupled mode where requests are released before responses sent) - does that pattern hold for python backend?

Reply:
If I am reading the code correctly, the scenario of returning cancelled on decoupled will work, because the `InferenceResponseSender` on decoupled takes the same `InferenceResponse` object that is used on non-decoupled scenario, and the cancelled status is set when the `InferenceResponse` object is constructed.

I will look into the cancelled query scenario on decoupled (and add a test case to show it works), because it is more worrying if the "requests are released before responses sent"

Reply:
Yeah, I think we probably need a `is_cancelled` method for `InferenceResponseSender` as well.

Reply:
Added a test case on decoupled model. The logic on request check is to check for cancellation on response factory pointer first. If it is undefined (i.e. non-decoupled case), then the cancellation is checked on the request pointer. If both response factory and request are undefined (should not happen), then an exception is throw.

Reply:
> @Tabrizian Can you elaborate more on why we want to check for cancellation on InferenceResponseSender?

It is not required to keep the inference request object alive. The use-case I had in mind is similar to the C++ backend APIs. In C Backend APIs, we have APIs for both `TRITONBACKEND_RequestIsCancelled` and `TRITONBACKEND_ResponseFactoryIsCancelled`. In Python backend response factory is similar to response sender.

As shown in the example [here](https://github.com/triton-inference-server/python_backend/blob/b136bf3fad77d5d467a6db5bd739be16eb9d9400/examples/decoupled/square_model.py#L201-L217) the request object may not be in scope when the request is being cancelled.

Reply:
Thanks for clarifying! Will update.

Reply:
Updated. The user is able to call either of the following:
```
request.is_cancelled()
request.get_response_sender().is_cancelled()  # does not depend on request after getting response sender
```